pkg/build/strategies/sti/sti.go

```diff
@@ -317,7 +317,6 @@ func (builder *STI) createBuildEnvironment() []string {
 	env, err := scripts.GetEnvironment(builder.config)
 	if err != nil {
 		glog.V(1).Infof("No user environment provided (%v)", err)
-		return nil
```

Comment:
i guess maybe since ConvertEnvironment has a named return value, "result" will be initialized and returned and the range loop will no-op since env is nil.

feels side-effecty to me, but i guess you are correct that it'll work.